### PR TITLE
EOS-18144: Modified motr mini provisioner according to new conf store keys

### PR DESCRIPTION
**Pull request by atulsdeshmukh2312**
…a logical disk group from confstore and configure using motr_setup

### CODE REVIEW

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 1]
```diff
@@ -101,7 +102,7 @@ def validate_file(file):
 def is_hw_node(self):
     try:
         node_type = Conf.get(self._index,
-                    f'cluster>{self._server_id}')['node_type']
+                             f'server_node')[machine_id]['type']
```
Comment (huanghua78) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:105]: Please fix this Codacy warning.
If there is no placeholder in a f-string, no need to have such a f-string. Plain string is enough.

Comment (huanghua78) [reply scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:105]: As I see, these changes are due to the changes in the conf schema. Am I right?

The schema should be defined carefully and not change it frequently.

Also, this "f'server_node')[machine_id]" id is used in multiple places. Can we define a function to get the "node", and use it everywhere?

**Comment by huanghua78** [comment]
This PR has a very long title. @atulsdeshmukh2312 , can you please refine the title?

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py (hunk 1 above)
Comment (atulsdeshmukh2312) [reply scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py]: Addressed review comments in e65902b..936a3f5

**Comment by huanghua78** [comment]
@atulsdeshmukh2312 How would this PR be tested?
Can you please test this PR on Jenkins?

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 2]
```diff
@@ -398,36 +408,32 @@ def check_pkgs(self, pkgs):
 def get_nids(self, nodes):
     """Get lnet nids of all available nodes in cluster."""
     nids = []
+    myhostname = execute_command(self, "hostname")[0].rstrip('\n')
```
Comment (yeshpal-jain-seagate) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:411]: use server_node>{MACHINE_ID}>hostname instead using $(hostname)

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 3]
```diff
@@ -62,27 +61,26 @@ def check_type(var, vtype, msg):
...
-
-def get_current_node(self):
-    """Get current node name using machine-id."""
+def get_machine_id(self):
     cmd = "cat /etc/machine-id"
     machine_id = execute_command(self, cmd)
     machine_id = machine_id[0].split('\n')[0]
-
     check_type(machine_id, str, "machine-id")
     if len(machine_id) != MACHINE_ID_LEN:
```
[the remark this answers is not in the document]
Comment (atulsdeshmukh2312) [reply scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py]: Done..88b0768..40d8664

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 4]
```diff
@@ -100,10 +98,10 @@ def validate_file(file):
```
[the remark this answers is not in the document]
Comment (atulsdeshmukh2312) [reply scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:95]: Done...40d8664..b18fe6b

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 5]
```diff
@@ -428,6 +425,33 @@ def get_nodes(self):
...
+            storage = node['storage']
+            cvg_count = storage['cvg_count']
+            for i in range(int(cvg_count)):
+                total_disks += len(storage["cvg"][i]["data_devices"])
+    return total_disks
+
+def check_data_disks_count(self):
+    total_disks = get_data_disks_count(self)
+    required_disks = get_data_parity_spare_count(self)
+    if (total_disks <= required_disks):
```
Comment (yeshpal-jain-seagate) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:442]: if variable name is required_disks, then total_disks <= required_disks sounds like a proper condition but we are rasising failure on this condition.

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 6]
```diff
@@ -428,6 +425,33 @@ def get_nodes(self):
...
+        raise MotrError(errno.EINVAL, f"Total disks of all nodes({total_disks})"
+                        f" must be >= (data+parity+spare)({required_disks}) disks")    
+    sys.stdout.write(f"Total disks={total_disks} and Required disks={required_disks}")
+
+def get_data_parity_spare_count(self):
+    total_disks = 0
+    cluster = list(Conf.get(self._index, 'cluster').values())[0]
+    total_disks += int(cluster["storage_set"][0]["data"])
+    total_disks += int(cluster["storage_set"][0]["parity"])
```
Comment (yeshpal-jain-seagate) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:451]: cluster>{cluster-id}>storage_set[N]>durability>parity

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 7]
```diff
@@ -428,6 +425,33 @@ def get_nodes(self):
...
+    total_disks += int(cluster["storage_set"][0]["spare"])
```
Comment (yeshpal-jain-seagate) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:452]: cluster>{cluster-id}>storage_set[N]>durability>spare as per the confluence

**Review by yeshpal-jain-seagate — APPROVED** [review]

**Review by madhavemuri — APPROVED** [review]

**Pull request merged**